eras/conway/impl/testlib/Test/Cardano/Ledger/Conway/ImpTest.hs

```diff
@@ -1,682 +1,375 @@
--- produced and consumed values and ensuring fees are correct.
-fixupTx ::
-  forall era.
-  ( EraTx era
-  , EraGov era
-  , ConwayEraTxBody era
-  , Signable
-      (DSIGN (EraCrypto era))
-      (Hash (HASH (EraCrypto era)) EraIndependentTxBody)
-  , TxOut era ~ BabbageTxOut era
-  ) =>
-  Tx era ->
-  ImpTestM era (Tx era)
-fixupTx tx = do
-  ImpTestState {impNES, impRootTxId, impRootTxCoin} <- get
-  let
-    utxos = impNES ^. nesEsL . esLStateL . lsUTxOStateL
-    pp = utxos ^. utxosGovStateL . curPParamsGovStateL
-  kpSpending <- lookupKeyPair =<< freshKeyHash
-  kpStaking <- lookupKeyPair =<< freshKeyHash
-  modify $ \st ->
-    st
-      { impRootTxId = TxId $ hashAnnotated (tx ^. bodyTxL)
-      }
-  let
-    outputsTotalCoin = sumAllCoin $ tx ^. bodyTxL . outputsTxBodyL
-    remainingCoin = impRootTxCoin <-> outputsTotalCoin
-    remainingTxOut =
-      BabbageTxOut @era
-        (mkAddr (kpSpending, kpStaking))
-        (inject remainingCoin)
-        NoDatum
-        SNothing
-  impRootTxCoinL .= remainingCoin
+    dRepAcceptedRatio @era
+      ratEnv
+      (gas ^. gasDRepVotesL)
+      (gasAction gas)
+
+-- | Calculates the ratio of CC members that have voted for the governance
+-- action
+calculateCommitteeAcceptedRatio :: forall era. ConwayEraGov era => GovActionId (EraCrypto era) -> ImpTestM era Rational
+calculateCommitteeAcceptedRatio gaId = do
+  eNo <- getsNES nesELL
+  RatifyEnv {reCommitteeState} <- getRatifyEnv
+  GovActionState {gasCommitteeVotes} <- lookupGovActionState gaId
+  ens <- getEnactState
   let
-    balancedTx =
-      setMinFeeTx pp tx
-        & bodyTxL . inputsTxBodyL %~ Set.insert (mkTxInPartial impRootTxId 0)
-        & bodyTxL . outputsTxBodyL %~ (remainingTxOut :<|)
-  wits <- mkTxWits (balancedTx ^. bodyTxL)
-  pure $ balancedTx & witsTxL .~ wits
-
--- | Constructs and submits a Conway era transaction. Takes a message that's
--- shown when the transaction fails and a function that constructs a transaction
--- from the basic transaction. The function does not have to balance the
--- produced and consumed values, this is done automatically.
-submitBasicConwayTx :: String -> (Tx Conway -> Tx Conway) -> ImpTestM Conway (TxId StandardCrypto)
-submitBasicConwayTx desc f = do
+    committee = ens ^. ensCommitteeL
+    members = foldMap' (committeeMembers @era) committee
+  pure $
+    committeeAcceptedRatio
+      members
+      gasCommitteeVotes
+      reCommitteeState
+      eNo
+
+-- | Logs the ratios of accepted votes per category
+logAcceptedRatio :: ConwayEraGov era => GovActionId (EraCrypto era) -> ImpTestM era ()
+logAcceptedRatio aId = do
+  dRepRatio <- calculateDRepAcceptedRatio aId
+  committeeRatio <- calculateCommitteeAcceptedRatio aId
+  logEntry "----- ACCEPTED RATIOS -----"
+  logEntry $ "DRep accepted ratio:\t\t" <> show dRepRatio
+  logEntry $ "Committee accepted ratio:\t" <> show committeeRatio
+  logEntry ""
```

Comment:
Why not add a function that can be applied to lists?
```suggestion
  logEntries 
    [ "----- ACCEPTED RATIOS -----"
    , "DRep accepted ratio:\t\t" <> show dRepRatio
    , "Committee accepted ratio:\t" <> show committeeRatio
    ]
```

Reply:
Or even better takes a special `header` argument and adds the dashes around it:
```haskell
logEntries header xs = ...
```
would result in: 
```
----- Header ------
```

Reply:
I think an even better solution would be to format logs with prettyprinter.

Reply:
Sure, that could be. We'll solve this problem at a different time. This was just a mere suggestion.

Reply:
I switched the logging type to `Doc ()`, this should add more flexibility to how we want to format the logs in the future.